src/tensor_ops/cmp/eq/mod.rs

```diff
@@ -0,0 +1,56 @@
+use crate::{
+    gradients::{NoneTape, Tape},
+    shapes::{Shape, Unit},
+    tensor::Tensor,
+};
+
+use super::{try_cmp_op, CmpKernel};
+
+mod cpu_kernel;
+#[cfg(feature = "cuda")]
+mod cuda_kernel;
+
+pub enum EqKernelOp {}
+
+impl<S: Shape, E: Unit, D: CmpKernel<EqKernelOp, E>, T: Tape<D>> Tensor<S, E, D, T> {
+    pub fn try_eq(self, other: Self) -> Result<Tensor<S, bool, D, NoneTape>, D::Err> {
+        try_cmp_op(self, other)
+    }
+
+    pub fn eq(self, other: Self) -> Tensor<S, bool, D, NoneTape> {
+        self.try_eq(other).unwrap()
+    }
+}
+
+pub fn eq<S: Shape, E: Unit, D: CmpKernel<EqKernelOp, E>, T: Tape<D>>(
+    lhs: Tensor<S, E, D, T>,
+    rhs: Tensor<S, E, D, T>,
+) -> Tensor<S, bool, D, NoneTape> {
+    lhs.eq(rhs)
+}
```

Comment:
Thoughts on making all the tensors reference instead of value? It wouldn't be consistent with other apis, but cmp isn't consistent in the first place since you can't backprop through it. Perhaps taking by reference would make that more clear (and reduce cloning where we can)?

```suggestion
impl<S: Shape, E: Unit, D: CmpKernel<EqKernelOp, E>, T: Tape<D>> Tensor<S, E, D, T> {
    pub fn try_eq(&self, other: &Self) -> Result<Tensor<S, bool, D, NoneTape>, D::Err> {
        try_cmp_op(self, other)
    }

    pub fn eq(&self, other: &Self) -> Tensor<S, bool, D, NoneTape> {
        self.try_eq(other).unwrap()
    }
}

pub fn eq<S: Shape, E: Unit, D: CmpKernel<EqKernelOp, E>, T: Tape<D>>(
    lhs: &Tensor<S, E, D, T>,
    rhs: &Tensor<S, E, D, T>,
) -> Tensor<S, bool, D, NoneTape> {
    lhs.eq(rhs)
}
```